### PR TITLE
Add gaussian perturbations and basic occlusion methods

### PR DESCRIPTION
**Pull request by thomas-mullor**
## Description

Code for :
 - word occlusion
 - token occlusion
 - gaussian noise perturbations
 - tests for linear interpolation perturbations
 
## Type of Change

- 🚀 New feature (non-breaking change which adds functionality)

## Checklist

- [x] I've read the [`CODE_OF_CONDUCT.md`](https://github.com/FOR-sight-ai/interpreto/blob/main/CODE_OF_CONDUCT.md) document.
- [x] I've read the [`CONTRIBUTING.md`](https://github.com/FOR-sight-ai/interpreto/blob/main/docs/contributing.md) guide.
- [x] I've successfully run the style checks using `make lint`.
- [] I've written tests for all new methods and classes that I created and successfully ran `make test`.
- [x] I've written the docstring in Google format for all the methods and classes that I used.

### CODE REVIEW

File: interpreto/attributions/aggregations/base.py [hunk 1]
```diff
@@ -17,10 +17,57 @@ class Aggregator(ABC):
...
+
+class SquaredMeanAggregator(Aggregator):
+    """
+    Square of mean of attributions
+    """
+
+    # TODO : remake this class with __method as a function chain of torch.mean and torch.square
+    # _method=torch.mean
+    def aggregate(self, results: torch.Tensor, _) -> Any:
+        return torch.mean(results, _) ** 2
```
Comment (AntoninPoche) [inline interpreto/attributions/aggregations/base.py]: The square mean aggregation for SquareGrad is the mean of the squared elements, not the square of the mean of the elements.

File: interpreto/attributions/perturbations/base.py [hunk 2]
```diff
@@ -9,69 +9,19 @@
...
-            if baseline.dtype != inputs.dtype:
-                raise ValueError(f"Baseline dtype {baseline.dtype} does not match expected dtype {inputs.dtype}.")
-        else:
-            raise TypeError("Baseline must be None, a float, or a PyTorch tensor.")
-
-        return baseline
-
     @abstractmethod
-    def perturb(self, item: ModelInput | Collection[ModelInput]) -> Collection[TokenEmbedding]:
+    def perturb(self, inputs: ModelInput | Collection[ModelInput]) -> Collection[TokenEmbedding]:
```
Comment (AntoninPoche) [inline interpreto/attributions/perturbations/base.py]: The output of the perturbation from the perturbator is not necessarily a `TokenEmbedding`; this is only the case for NLP perturbators.

Furthermore, the `perturb` method outputs 2 elements (the perturbed elements and the mask). Hence, this should be reflected in the typing. (I may have missed an update on this).

File: interpreto/attributions/perturbations/occlusion.py
Comment (AntoninPoche) [inline interpreto/attributions/perturbations/occlusion.py:1]: I find the name of the file `occlusion.py` misleading, I would prefer `occlusion_perturbations.py`.

Comment (AntoninPoche) [reply interpreto/attributions/perturbations/occlusion.py:1]: Shouldn't there be an `occlusion.py` file implementing the Occlusion method? in `attributions.methods`?

File: interpreto/attributions/perturbations/occlusion.py [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+    def __init__(
+        self, tokenizer: PreTrainedTokenizer, inputs_embeddings: torch.nn.Module, mask_value: str | None = None
+    ):
+        # TODO : Currently only deals with Huggingface PreTrainedTokenizer (or equivalents), should be more general
+        self.tokenizer = tokenizer
+        self.inputs_embeddings = inputs_embeddings
+        self.mask_value = mask_value or tokenizer.mask_token
+
+    @singledispatchmethod
+    def perturb(self, inputs) -> tuple[torch.Tensor, torch.Tensor]:
```
Comment (AntoninPoche) [inline interpreto/attributions/perturbations/occlusion.py]: In the case of single dispatch methods, where is supposed to be the doc string?

File: interpreto/attributions/perturbations/occlusion.py [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+        ]
+        # Get words embeddings for each variation
+        embeddings = torch.stack([self.inputs_embeddings(torch.tensor(variation)) for variation in variations])
+        # Return embeddings and identity matrix as mask
+        return embeddings, torch.eye(n_perturbations)
+
+    @perturb.register(Iterable)
+    def _(self, inputs: Iterable):
+        # Perturb a batch of inputs (or nested batchs of inputs)
+        return [self.perturb(input) for input in inputs]
```
Comment (AntoninPoche) [inline interpreto/attributions/perturbations/occlusion.py]: You were supposed to bring balance to the Force, not leave it in darkness!

File: interpreto/attributions/perturbations/occlusion.py [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+        # Get tokens from str input
+        tokens = self.tokenizer.tokenize(inputs)
+        n_perturbations = len(tokens)
+        # Create variations by masking each token
+        variations = [
+            self.tokenizer.convert_tokens_to_ids(tokens[:i] + [self.mask_value] + tokens[i + 1 :])
+            for i in range(n_perturbations)
+        ]
+        # Get words embeddings for each variation
+        embeddings = torch.stack([self.inputs_embeddings(torch.tensor(variation)) for variation in variations])
```
Comment (AntoninPoche) [inline interpreto/attributions/perturbations/occlusion.py]: Did you check the shape of the resulting embeddings? `self.inputs_embeddings` might output a batch dimension even for single inputs.

File: tests/attributions/methods/test_perturbators.py
Comment (AntoninPoche) [inline tests/attributions/methods/test_perturbators.py:1]: I think this file should be in `tests/attributions/perturbations/`

File: tests/attributions/methods/test_perturbators.py [hunk 6]
```diff
@@ -0,0 +1,75 @@
+import pytest
+import torch
+
+from interpreto.attributions.perturbations.base import GaussianNoisePerturbator
+from interpreto.attributions.perturbations.linear_interpolation_perturbation import (
```
Comment (AntoninPoche) [inline tests/attributions/methods/test_perturbators.py]: These perturbations should be added to the `perturbations.__init__.py` so that all perturbations can be imported from the same module.

File: tests/attributions/methods/test_perturbators.py
Comment (AntoninPoche) [reply tests/attributions/methods/test_perturbators.py:1]: There should be a common test for all perturbations where we iterate on all perturbations.

File: tests/attributions/methods/test_perturbators.py [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+
+
+def test_gaussian_noise_perturbator_perturb():
+    perturbator = GaussianNoisePerturbator(n_perturbations=3, std=0.1)
+    inputs = torch.tensor([[1.0, 2.0], [3.0, 4.0]])
+    perturbed_inputs, _ = perturbator.perturb(inputs)
+
+    assert perturbed_inputs.shape == (3, 2, 2)
+    assert perturbed_inputs.device == inputs.device
+    assert torch.allclose(perturbed_inputs.mean(dim=0), inputs, atol=0.2)
```
Comment (AntoninPoche) [inline tests/attributions/methods/test_perturbators.py]: This seems like a test that could go wrong in some cases. Even with an `std` of 0.1, the probability of a difference higher than 0.2 is not zero.

Comment (AntoninPoche) [reply tests/attributions/methods/test_perturbators.py]: Tests which can go wrong depending on the seed are really hard to debug.

File: interpreto/attributions/perturbations/occlusion.py (hunk 3 above)
Comment (thomas-mullor) [reply interpreto/attributions/perturbations/occlusion.py]: In the generic function
